python-sdk/src/astro/databases/base.py

```diff
@@ -362,7 +364,7 @@ def create_schema_and_table_if_needed(
         is_file_pattern_based_schema_autodetection_supported = (
             self.check_file_pattern_based_schema_autodetection_is_supported(source_file=file)
         )
-        if self.schema_exists(table.metadata.schema) and if_exists == "replace":
+        if if_exists == "replace":
```

Comment:
@kaxil is this safe? Do we have any documentation as to why we check if the schema exists first?

Reply:
This was only added in https://github.com/astronomer/astro-sdk/commit/730d273b32e4f6a453f90837942a8490bc766cf9 last week and I don't think we need it -- drop query is idempotent. And if the schema doesn't exist it should "error" out rather than ignoring